Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -88,21 +207,91 @@ public struct SwiftlyHTTPClient {
             return release
         }
 
-        return try await self.mapGitHubTags(limit: limit, filterMap: filterMap) { page in
-            try await self.getReleases(page: page)
+        guard !swiftOrgFiltered.isEmpty else {
+            return []
+        }
+
+        swiftOrgFiltered.sort(by: >)
+
+        return if let limit = limit {
+            Array(swiftOrgFiltered[0..<limit])
+        } else {
+            swiftOrgFiltered
         }
     }
 
     /// Return an array of Swift snapshots that match the given filter, up to the provided
     /// limit (default unlimited).
-    ///
-    /// TODO: retrieve these directly from swift.org instead of through GitHub.
     public func getSnapshotToolchains(
+        platform: PlatformDefinition,
+        arch a: String? = nil,
```

Comment:
Is this arch variable just for testing? If so, then it might be better to add a method to `Platform` to retrieve the arch, which can be mocked globally rather than just in this individual function.

Reply:
Right, this allows the HTTPClientTests to be able to test for all os/arch, independently from the platform that is currently running the test. I don't think that we need a mock for it since most of the tests rely on the current platform, and could run into problems if they try to install a toolchain for an alternate platform.